redfish-core/lib/pcie.hpp

```diff
@@ -251,15 +255,18 @@ static inline void
 }
 
 /**
- * @brief Main method for adding Link Status to the requested device
- * @param[in,out]   asyncResp       Async HTTP response.
- * @param[in]       pcieDevice      PCIe device name/ID.
+ * @brief Helper method to find the PCIe slot path
+ *
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       pcieDevice  PCIe device name/ID.
+ * @param[in]       callback    Callabck method.
  */
-static inline void addLinkStatusToPcieDevice(
-    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-    const std::string& pcieDevice)
+template <typename Callback>
+inline void
+    findPcieSlotPath(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                     const std::string& pcieDevice, Callback&& callback)
 {
-    auto respHandler = [asyncResp, pcieDevice](
+    auto respHandler = [asyncResp, pcieDevice, callback](
```

Comment:
Can we change to
[asyncResp, pcieDevice, **callback{std::move(callback)}**]( ...

In bmcweb, function pointers are cast to rvlaues. 
I think we do it to avoid a copy constructor call, but i'm not sure